### PR TITLE
Use pointer receivers instead of value receivers.

### LINKED ISSUE #48: Method receivers are not using pointers causing a lot of unnecessary copying.
Is there a reason all methods on the structs are copying the struct instead of using pointer receivers?

### PR DESCRIPTION
**Pull request by tw1nk**
Fixes #48 
Also implement tags from @bradleyfalk as well as a member function for Permanently delete members.

### CODE REVIEW

**Comment by zeekay** [comment]
Looks good, I'll review this weekend! Thanks for your commits!

**Comment by tw1nk** [comment]
Oh, I should probably have made a separate branch for this pull request. 

I had a need to get the tags functionality in as well so I merged @bradleyfalk changes into my master branch. And I also needed the Delete Permanent function, so I implemented it. 

If you want I could probably create a separate branch for this.

**Comment by zeekay** [comment]
I think we just need to test it -- make sure there are no issues switching to pointer receivers.

File: go.mod [hunk 1]
```diff
@@ -1,4 +1,4 @@
-module github.com/hanzoai/gochimp3
+module github.com/tw1nk/gochimp3
```
Comment (boyan-improbable) [inline go.mod:1]: Was this supposed to be merged, @zeekay @tw1nk?

Comment (zeekay) [reply go.mod:1]: Thanks for catching that!

Comment (tw1nk) [reply go.mod:1]: Sorry about that, haven't used GitHub for a couple of years and thought it didn't update pull requests unless you actively did an update on the webpage.

**Pull request merged**